schemas/announcement_schema.py

```diff
@@ -0,0 +1,43 @@
+# schemas/announcement_schema.py
+from pydantic import BaseModel, root_validator, model_validator, ConfigDict
+from typing import Optional
+from datetime import datetime
+
+
+class AnnouncementCreate(BaseModel):
+    id: Optional[int] = None
+    title: str
+    description: str
+    announcement_date: datetime
+    announcement_deadline: Optional[datetime] = None
+    image: Optional[str] = None
+
+    class Config:
+        orm_mode = True
+        from_attributes = True
+
+
+class AnnouncementUpdate(BaseModel):
+    id: int
```

Comment:
I do not think we should allow people to change the id

Reply:
I think adding the id over there is a way to say we should have the id of the announcement to be updated! But your point made me add a condition to ignore the id if the client attempts to update it. Thank you!